### DISCUSSION TITLE
[Feature]: photo list should show if an item is in an album

### DISCUSSION THREAD

**Discussion by JeromeGillard** [immich-app/immich#1621] [Feature Request]
### Feature detail

Instead of going within each of my pictures one by one, it would be nice if the `/photos` screen could show me at a glance if a picture is already in an album or not.

Suggestion: Maybe by having an icon or the album name displayed as an overlay on mouse hover?

### Platform

Web

**Comment by JeromeGillard** [comment]
Discord `SMAW`:
> If you go to the shared album and click add photos you see which are already added... In the overview

That is indeed convenient. And very well implemented from an UX point of view. Kudos for it!

However those extra steps of opening the album (slow) then click add are quite opposed to having it "at a glance" in the photo list.

**Comment by alextran1502** [comment]
~Do you have a proposal?~
Nvm see it

**Comment by nullket** [comment]
I have the same issue. I simply like to know if a photo is not in an album at all. This helps me to organize my photos by forcing me to put photos in an album (or delete unwanted ones). Otherwise I would simply overlook so many "minor" photos in my full timeline. 

A proposal on how this could be solved: On the original macOS Photo App I simply created a "smart album" which listed photos that are not in any other album. Maybe something like "smart albums" (including filters, detected objects/categories) could be something for Immich as well. This would improve the album functionality and also solve this issue here?

**Comment by alextran1502** [comment]
> I have the same issue. I simply like to know if a photo is not in an album at all. This helps me to organize my photos by forcing me to put photos in an album (or delete unwanted ones). Otherwise I would simply overlook so many "minor" photos in my full timeline.
> 
> A proposal on how this could be solved: On the original macOS Photo App I simply created a "smart album" which listed photos that are not in any other album. Maybe something like "smart albums" (including filters, detected objects/categories) could be something for Immich as well. This would improve the album functionality and also solve this issue here?

This feature can be encompassed into a filter/search system as well.

**Comment by rriley** [comment]
Hi,

I was just wondering if there is a way to do this yet.  I'm interested in getting started with Immich, and it would be really helpful to me to be able to see which files haven't been added to an album.  (I plan to organize the vast majority of my images into albums.)

I would be fine with any way to do it: A smart album, a special search (which feels like the best way to do it), etc.

**Comment by Sherlock1979** [comment]
#2559 is a similar request from me. Here is a copy-paste so that you can see my take on it:

> It would be great to be able to view all assets that are not put in any albums yet.
> 
> It would help organise the photos. I would appreciate this feature a lot and it would help my categorisation and family photo sharing workflow.
> 
> Maybe it can be a search option or it could be accessed from the 'Library' section? We can call it maybe 'Uncategorised'
> 
> (BTW Photoprism has it as a menu item within albums and calls it: 'unsorted' ?!)
> 
> Thanks for considering it.

**Comment by Freetodream** [comment]
Any update on this? Just started with Immich and started to organize my photos, but realized there is no easy way to find which photos are not in an album. Would love to see this feature in the upcoming builds.

**Comment by piotrkustal** [comment]
Would also appreciate to have this filter/search enabled to filter-out/show which photos are not assigned to any of albums.

Simple scenario:
* I'm uploading hundreds of photos each day
* some of them are assigned into albums
* some not
* for those which are not I'd like to filter out and mass-delete or move to "undefined" album

Thank you.

**Comment by varpness** [comment]
I would very much like to see this as well. Often I, or other users, upload images which were taken before photos previously uploaded and it's difficult to find them in the mix. Especially with people who share photos, we need a way to quickly see if photos haven't been shared yet so they don't get lost.

**Reply by undaunt** [reply]
Agreed, I've been doing CLI import with specific subfolders that are named, using the album flag, and that works great. However, I had one big prior export that was already nested into subfolders for years, months, days, etc. so I didn't want 365 new albums, but now I need to locate all of these prior imported photos somehow to appropriate add them to albums.

**Comment by jstrot** [comment]
+1

I've never had a photo library app before so I'm doing an initial import of several decades of photos and movies that are, for the most part, uncategorized if only by date.

I'd really like to be able to view all items not part of any album, still sorted by date, so I can efficiently categorize them.

An other thought: Some photos may not end up in any album (a "cute bunny" album with a single photo would be overkill). Perhaps having a sort of "review flow" where newly uploaded items (and not assigned to an album during upload) could be marked with some flag (e.g., "new" or "uncategorized") until it is cleared or the item is assigned to an album.

**Reply by halsafar** [reply]
Looking for a "review flow" is exactly what lead me here.  

I like to let my phone upload for a few months and then go classify them or delete.  Not being able to find photos by "not in an album" breaks the only review flow I know.

**Comment by ahatdude** [comment]
+1

This would definitely be a great feature to allow us to keep our photo libraries organized with albums.
Just did a full import and I have some videos and pictures that have wrong dates which I'm trying to organize into the right albums, however when going through it I sometimes can't recall if I already added a certain picture or not.
It would also be nice to be able to quickly see what albums the pictures are in - for example a general album for New York but a specific one for pictures at Central Park. In this case it'd make sense, but what if I added one of the Central Park pictures to another album from a park in Maryland by mistake, it would be a simple way to see everything and correct the mistake.

**Comment by p3t3r** [comment]
+1

Also like  a review flow like mentioned by jstrot would be great. 

Now I upload all my finished camera photos into Immich, but also mobile photos that I want to review and delete later.

**Comment by halsafar** [comment]
Immich 1.95.1 solves this for me by allowing Advanced Search -> Display Options -> Not In Any Album

**Reply by theone11** [reply]
The problem is that in search results there is no timeline scrollbar and no date seperators - I opened a feature request for it #8523 or should I open a bug?

**Reply by justin13888** [reply]
@halsafar's solution is good for me. Just thinking if there needs to be any explicit button like PhotoPrism?

![331746302-72e3201b-895d-4ed0-9a72-9954fdd4c113](https://github.com/immich-app/immich/assets/20733699/1bcae257-c6af-489b-a86f-81e44e2250cb)

**Reply by theone11** [reply]
The problem is that there are no date separators and no time line in this option.

**Reply by kosli** [reply]
"Not In Any Album" does not work for me as the Android App sync with Album names, e.g. "Camera", "WhatsApp Images" etc -> so it would be great if these could be filtered out as well like "Not in Any Album nor in "Camera etc.."")

**Comment by nothing2obvi** [comment]
The "Not in any album" option is nice, but those photos in Search aren't laid out as nicely as in the timeline, and you can't multi-select by clicking on one asset, holding shift, then clicking on the last asset of the set (at least I can't). You have to do it one-by-one.

Plus, seeing an in-album indicator (just a see-through icon) on assets in the timeline would give peace of mind as you scroll through your timeline. You'd know that every asset is accounted for (i.e. placed in an album). This is like @JeromeGillard's original idea.

**Comment by C-Otto** [comment]
I hacked my personal instance to only show assets in the main view if they are not assigned to any album. This retains the date separators and other features that are not avaible from the search result view. However, with this approach I do not see the (related?) assets that already are in some album.

Would this already help, or should the solution show both types of assets in the same view (in-album and not-in-album)?

If both are shown in the same view, I fear that it might be too hard to see the difference. If one made the visual difference more outstanding, this would cause the view to be unsuitable for generic browsing - calling for a new view as suggested before.

1. The main view can be configured to only show assets which are not in any album. With this, one would benefit from having the timeline, but one would not be able to see the related photos in the same view.
2. The main view is COPIED to another view which only shows assets which are not in any album. This corresponds to the "Unsorted" view mentioned above. With this, one benefits from having the timeline, AND one would be able to see related assets in the main view - however, one would need to toggle between both views, e.g. by opening two browser tabs.
3. The main view is changed so that assets that are not in any album are easy to identify. For this, one could add an icon on top of the preview images (similar to the video icon in the upper right, or maybe much larger spanning the whole preview), scale images down so that they are visibly smaller (indicating: already in an album), or play around with color tints.
4. The search result view is updated so that it also shows the timeline.

**Reply by bo0tzz** [reply]
4 would be the best/most general solution, but would require adding sorting options for the search which is pretty complex.

**Comment by proofrock** [comment]
This is the single thing that is still keeping me on Photoprism.

**Comment by proofrock** [comment]
As a (ugly) workaround, I found that this works:

- Create an album to receive the "not in any album" pics (e.g. `Unsorted`)
- Access the postgres database with something like
```bash
docker exec -it <postgres_container_name> psql -U <DB_USERNAME> -d <DB_DATABASE_NAME>
```
- Execute the following SQL statement to assign all the assets that aren't assigned to any album to the `Unsorted` album:
```sql
INSERT INTO albums_assets_assets ("assetsId", "albumsId")
SELECT a.id AS "assetsId", a2.id AS "albumsId"
  FROM assets a 
  JOIN albums a2 ON a2."albumName" = 'Unsorted'
 WHERE NOT EXISTS (SELECT 1 FROM albums_assets_assets aaa WHERE aaa."assetsId" = a.Id)
   AND a."originalPath" LIKE '/libraries/%';
```
(Omit the first line if you just want to see whether there are unsorted images)

I can't say it's a perfectly correct solutions, but it works. Similarly it's possible to assign photos to albums based on filesystem path.

Edit: added the last line of the SQL to differentiate from auto-converted videos, check that `/libraries/` is your import folder structure.

**Comment by MD77MD** [comment]
Suggestion:  the album name displayed as an overlay on mouse hover

make it optional to show picture info like
1.tages
2. album
3. star count
4. etc

and the ability to choose how its shown 
1. mouse hover on picture 
2. permanent on picture 
3. disabled (by default)

**Comment by EnricoBara** [comment]
I think the mouse over in mobile can't be easily reproduced.

Maybe an easier solution could be a tick at the right of the album in the 'add to album' tab, where the asset is already in. Same in web app.

This also could be useful to remove an asset from an album added by mistake without entering in the album.

Also, to see if an asset is into some album, could be put a line in its info section.

[Example](https://github.com/user-attachments/assets/8318fe43-a75a-46ef-8d9f-4292d4e01386)

**Comment by bo0tzz** [comment]
#21029

**Comment by markhavemann** [comment]
+1

**Comment by Zepherian04** [comment]
Could behave like Spotify's "Add to Playlist" feature where you click the option and it has a checkmark next to all playlists the song is in, and no checkmark next to those it's not
